src/tests/ftest/io/ior_single_server.py

```diff
@@ -24,114 +24,116 @@
-
-        self.agent_sessions = agent_utils.run_agent(self.basepath,
-                                                    self.hostlist_servers,
-                                                    self.hostlist_clients)
-        server_utils.run_server(self.hostfile_servers, self.server_group,
-                                self.basepath)
+        """Set up each test case."""
+        # Start the servers and agents
+        super(IorTestBase, self).setUp()
+
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file.write_host_file(
+            self.hostlist_clients, self.workdir, None)
+
+        # Get the parameters for IOR
+        self.ior_cmd = IorCommand()
+        self.ior_cmd.set_params(self)
+        self.processes = self.params.get("np", '/run/ior/client_processes/*')
 
     def tearDown(self):
+        """Tear down each test case."""
         try:
-            if self.hostfile_clients is not None:
-                os.remove(self.hostfile_clients)
-            if self.hostfile_servers is not None:
-                os.remove(self.hostfile_servers)
             if self.pool is not None and self.pool.attached:
                 self.pool.destroy(1)
         finally:
-            if self.agent_sessions:
-                agent_utils.stop_agent(self.agent_sessions,
-                                       self.hostlist_clients)
-            server_utils.stop_server(hosts=self.hostlist_servers)
+            # Stop the servers and agents
+            super(IorTestBase, self).tearDown()
 
-    def test_singleserver(self):
-        """
-        Test IOR with Single Server config.
+    def execute_ior(self, ior_flags=None, object_class=None):
+        """Execute ior with optional overrides for ior flags and object_class.
 
-        :avocado: tags=ior,singleserver
+        If specified the ior flags and ior daos object class parameters will
+        override the values read from the yaml file.
+
+        Args:
+            ior_flags (str, optional): ior flags. Defaults to None.
+            object_class (str, optional): daos object class. Defaults to None.
         """
+        # Get the parameters used to create a pool
+        mode = self.params.get("mode", "/run/pool/*")
+        uid = os.geteuid()
+        gid = os.getegid()
+        group = self.params.get("setname", "/run/pool/*", self.server_group)
+        scm_size = self.params.get("scm_size", "/run/pool/*")
+        nvme_size = self.params.get("nvme_size", "/run/pool/*", 0)
+        svcn = self.params.get("svcn", "/run/pool/*", 1)
+
+        # Initialize a python pool object then create the underlying
+        # daos storage
+        self.pool = DaosPool(self.context)
+        self.pool.create(
+            mode, uid, gid, scm_size, group, None, None, svcn, nvme_size)
+
+        # Initialize MpioUtils if IOR is running in MPIIO mode
+        if self.ior_cmd.api.value == "MPIIO":
```

Comment:
We actually want to use mpich for both DAOS and MPIIO modes.  The patch I am working on to get ior working in CI is doing this successfully.

Reply:
I think this needs to stay.   The IOR cmdline needs the "-a" parameter that specifies which API should be used for IOR (currently daos or mpiio)    If the -a is not used the default api is posix.

Reply:
Yes, sorry I was not more clear.  I'm not saying that ior shouldn't be given `-a {MPIIO|DAOS}` just that the use of two different MPI libraries for each can go away.